src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,624 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.NetCore.Analyzers.Performance;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferHashDataOverComputeHashAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpPreferHashDataOverComputeHashFixer>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferHashDataOverComputeHashAnalyzer,
+    Microsoft.NetCore.VisualBasic.Analyzers.Performance.BasicPreferHashDataOverComputeHashFixer>;
+
+namespace Microsoft.CodeAnalysis.NetAnalyzers.UnitTests.Microsoft.NetCore.Analyzers.Performance
+{
+    public class PreferHashDataOverComputeHashTests
+    {
+        [Fact]
+        public async Task CSharpCreateHelperChainCase()
+        {
+            string csInput = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        byte[] digest = {|#0:SHA256.Create().ComputeHash({|#1:buffer|})|};
+    }
+}
+";
+
+            string csFix = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        byte[] digest = SHA256.HashData(buffer);
+    }
+}
+";
+
+            await TestCS(
+                csInput,
+                csFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1));
+        }
```

Comment:
You could wrap the methods that use the same argument by putting the whole call into a helper method.